### PR TITLE
Add support for Windows Subsystem Linux - refactored

### PR DESCRIPTION
**Pull request by bzoz**
Continued from https://github.com/Microsoft/vscode-node-debug2/pull/126. Refactored like https://github.com/Microsoft/vscode-node-debug/pull/158., to add WSL support for `inspector` protocol.

Add new "useWSL" launch attribute. When used, Bash on Windows will be used to launch runtimeExecutable. Source files directory mapping will also be automatically added.

/cc @weinand

### CODE REVIEW

**Comment by alanosman** [comment]
When will this make it's way into the insiders build?

**Comment by yodurr** [comment]
Are you asking when with the WSL support for this arrive on Windows 10 insiders?  If yes, you can use the current retail Windows 10 (creators update) for this or any insiders release since.  No recent WSL work was required to support this.

**Comment by alanosman** [comment]
@yodurr I am asking "when can we debug a node app running on WSL from Visual Code". i.e. set a breakpoint in your node code, press F5 and start debugging, but with the process launching inside bash,  Currently, you can only attach to a bash node process through a port.  It would be nice to launch and go right into debugging.

**Comment by Alkasai** [comment]
@yodurr I think @alanosman was referring to the VSCode Insiders.

**Comment by roblourens** [comment]
This is in node-debug ("legacy") but not node-debug2 ("inspector"). @weinand does this PR match what's in node-debug now?

**Comment by bzoz** [comment]
@roblourens yes, this reimplementation of the method that was added to `vscode-node-debug`

**Comment by yodurr** [comment]
Thanks for clarifying.  Agree @alanosman the scenario you describe is what we're excited about seeing supported.  This PR makes it happen.

**Comment by alanosman** [comment]
@yodurr so, just so I am clear, this is not in the insiders build yet, right?

File: src/nodeDebugAdapter.ts [hunk 1]
```diff
@@ -68,8 +69,17 @@ export class NodeDebugAdapter extends ChromeDebugAdapter {
 
         const port = args.port || utils.random(3000, 50000);
 
+        if (args.useWSL && !wsl.subsystemLinuxPresent) {
```
Comment (roblourens) [inline src/nodeDebugAdapter.ts]: `subsystemLinuxPresent` is a function

**Comment by bzoz** [comment]
Updated

**Comment by roblourens** [comment]
Thanks! I'll publish this and pull it into the vscode build so it's available in the next Insiders build. Then it will be in stable 1.18, the first week in November.

**Comment by roblourens** [comment]
FYI @weinand I took this and found some tweaks in node-debug, and copied your fix from https://github.com/Microsoft/vscode-node-debug/commit/3a0f1a860b22ee7af9e14a9a0b09961e30b7aa10#diff-2a53bfce93d1a0ef575c1c6581a6cb1b. LMK if you can think of anything else I need to change.

**Pull request merged**